src/vcf/validator.cpp

```diff
@@ -132,6 +132,65 @@ namespace ebi
         }
     }
 
+    bool is_compressed_file(std::istream &input,
+                            const std::string &source)
+    {
+        std::string message = "Input file is a compressed file ";
+
+        if (source != ebi::vcf::STDIN) {
+            boost::filesystem::path source_name(source);
+            std::string file_extension = source_name.extension().string();
+
+            if (file_extension == ".zip") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(zip)";
+                return true;
+            }
+            else if (file_extension == ".tar") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar)";
+                return true;
+            }
+            else if (file_extension == ".gz") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar.gz)";
+                return true;
+            }
+            else if (file_extension == ".xz") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar.xz)";
+                return true;
+            }
+            else if (file_extension == ".rar") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(rar)";
+                return true;
+            }
+        }
+
+        unsigned char magic[10];
```

Comment:
until here, the check is done with the extension, and from here to the end it checks the magic numbers. both parts should be extracted to functions, and `is_compressed_file` therefore should contain just calls to those two functions.